Create fav.js

yrrah2 opened the pull request:
I am back with a much more user friendly favorite boards customization.

This time you can drag and press buttons to rearrange, add to or delete boards.

Here is the tab with no favorites
![noboards](https://cloud.githubusercontent.com/assets/9469469/6549152/21bc38d8-c673-11e4-92bc-2a133856b4b4.png)

Here is the tab after I used the text box to add in the best boards of the website (obviously)
![bestboards](https://cloud.githubusercontent.com/assets/9469469/6549162/35a18074-c673-11e4-9cf5-0fd90449a8aa.png)

Here is the tab after I used the minus buttons to remove boards I didn't want
![deletingboards](https://cloud.githubusercontent.com/assets/9469469/6549165/4b2fd04e-c673-11e4-9ab6-41a92e5e61dc.png)

Here is the tab after I used drag and drop to rearrange the boards
![movingboards](https://cloud.githubusercontent.com/assets/9469469/6549170/618b46b6-c673-11e4-8eb6-51c1bfc8aa59.png)

And finally this is the tab after I added in boards a-z
![lotsofboards](https://cloud.githubusercontent.com/assets/9469469/6549173/6f68e612-c673-11e4-8423-da44c8257f66.png)


The script assumes the user isn't a complete idiot that tries to add non-existent boards, but it is easy to remove such boards and one of the pre-existing scripts on here already fixes that.

yrrah2 commented:
Okay, I removed that line.

Thanks.

marktaiwan commented:
You'll also want to enclose the whole thing in `$(document).ready` so that your variables won't be in the global namespace

yrrah2 commented:
I think you will find you you drag one board over another, the list will move down.

As for the second point I am working on that now.

marktaiwan commented:
I know the list is draggable, but the change doesn't appear to persist after page refresh, if you simply rearranged them without adding or removing any boards.

yrrah2 commented:
You are right, I don't know how I didn't notice this.

I had it working some time ago but removed it thinking that the localStorage.favorites was updating as the user changed the order as it was when you added and removed boards.

I also made the generate list function no longer a variable despite it returning.

I have fixed these issues now.

ctrlcctrlv commented:
why was this closed?

yrrah2 commented:
I closed it because the variables were all global and the script wont work with them being made local currently.
